### ISSUE TITLE
Epic 6: Advanced Model Training

### ISSUE THREAD

**Issue by Hlumelo-Ncedana** [Hlumelo-Ncedana/hospital-readmission-prediction#6]
As a data scientist, I want to train advanced models so that I can achieve better prediction performance

Tasks:

Implement Random Forest classifier

Train XGBoost model

Perform hyperparameter tuning with GridSearchCV

Evaluate models on validation set

Compare all model performances
